crates/lair_keystore_api/src/lair_api/derive_seed.rs

```diff
@@ -0,0 +1,103 @@
+use super::*;
+
+/// Get entry_info for an entry by tag from lair.
+#[derive(Debug, serde::Serialize, serde::Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct LairApiReqDeriveSeed {
+    /// Msg id to relate request / response.
+    pub msg_id: Arc<str>,
+
+    /// Tag of existing seed.
+    pub src_tag: Arc<str>,
+
+    /// If source seed is deep-locked, this passphrase will unlock it.
+    #[serde(skip_serializing_if = "Option::is_none", default)]
+    pub src_deep_lock_passphrase: Option<DeepLockPassphraseBytes>,
+
+    /// Tag under which to store derived seed.
+    pub dst_tag: Arc<str>,
+
+    /// Optional passphrase with which to deep-lock derived seed.
+    #[serde(skip_serializing_if = "Option::is_none", default)]
+    pub dst_deep_lock_passphrase: Option<DeepLockPassphraseBytes>,
+
+    /// The hierarchy of nested derivations.
+    /// For instance, [0, 2, 1] would specify that the destination seed is the
+    /// 2nd derivation of the 3rd derivation of the 1st derivation of the source seed.
```

Comment:
```suggestion
    /// For instance, [99, 1, 42] would specify that the destination seed is the
    /// 42nd derivation of the 1st derivation of the 99th derivation of the source seed.
```

Reply:
oh, i see, you were trying to indicate the zero-based-ness of it. I just thought you got the numbers wrong : )

Well, I'll leave my comment up. I'm not sure we need to be pedantic about the ordinality. Feel free to take or leave this.